docker-compose.yml

```diff
@@ -1,48 +1,63 @@
 version: '3'
 
 services:
-  orbs-network-node_1:
+  orbs-network-node-1:
     image: orbs:export
     environment:
       NODE_NAME: node1
       PORT: 8080
       GOSSIP_PORT: 4400
       GOSSIP_PEERS: 192.168.199.3,192.168.199.4
-      LEADER: 'true'
+      NODE_PUBLIC_KEY: dfc06c5be24a67adee80b35ab4f147bb1a35c55ff85eda69f40ef827bddec173
+      NODE_PRIVATE_KEY: 93e919986a22477fda016789cca30cb841a135650938714f85f0000a65076bd4dfc06c5be24a67adee80b35ab4f147bb1a35c55ff85eda69f40ef827bddec173
+      CONSENSUS_LEADER: dfc06c5be24a67adee80b35ab4f147bb1a35c55ff85eda69f40ef827bddec173
     ports:
       - 8080:8080
     networks:
       orbs-network:
         ipv4_address: 192.168.199.2
 
-  orbs-network-node_2:
+  orbs-network-node-2:
     image: orbs:export
     environment:
       NODE_NAME: node2
       PORT: 8080
       GOSSIP_PORT: 4400
       GOSSIP_PEERS: 192.168.199.2,192.168.199.4
-      LEADER: 'false'
+      NODE_PUBLIC_KEY: 92d469d7c004cc0b24a192d9457836bf38effa27536627ef60718b00b0f33152
+      NODE_PRIVATE_KEY: 3b24b5f9e6b1371c3b5de2e402a96930eeafe52111bb4a1b003e5ecad3fab53892d469d7c004cc0b24a192d9457836bf38effa27536627ef60718b00b0f33152
+      CONSENSUS_LEADER: dfc06c5be24a67adee80b35ab4f147bb1a35c55ff85eda69f40ef827bddec173
     ports:
       - 8081:8080
     networks:
       orbs-network:
         ipv4_address: 192.168.199.3
 
-  orbs-network-node_3:
+  orbs-network-node-3:
     image: orbs:export
     environment:
       NODE_NAME: node3
       PORT: 8080
       GOSSIP_PORT: 4400
       GOSSIP_PEERS: 192.168.199.2,192.168.199.3
-      LEADER: 'false'
+      NODE_PUBLIC_KEY: a899b318e65915aa2de02841eeb72fe51fddad96014b73800ca788a547f8cce0
+      NODE_PRIVATE_KEY: 2c72df84be2b994c32a3f4ded0eab901debd3f3e13721a59eed00fbd1da4cc00a899b318e65915aa2de02841eeb72fe51fddad96014b73800ca788a547f8cce0
+      CONSENSUS_LEADER: dfc06c5be24a67adee80b35ab4f147bb1a35c55ff85eda69f40ef827bddec173
     ports:
       - 8082:8080
     networks:
       orbs-network:
         ipv4_address: 192.168.199.4
 
+  orbs-e2e:
+    image: orbs:e2e
+    environment:
+      API_ENDPOINT: http://orbs-network-node-3:8080/api/
+    command: sh -c "sleep 5 && ./e2e.test"
```

Comment:
this is ok for now, but sooner rather than later we need to implement some sort of startup_test mechanism which returns true if all services finished their init mechanism (and can maybe block until that)

Reply:
Agreed.